moveit_core/robot_trajectory/src/robot_trajectory.cpp

```diff
@@ -264,20 +285,38 @@ void RobotTrajectory::getRobotTrajectoryMsg(moveit_msgs::RobotTrajectory& trajec
       trajectory.joint_trajectory.points[i].positions.resize(onedof.size());
       trajectory.joint_trajectory.points[i].velocities.reserve(onedof.size());
 
+      // Compute torques if needed (no effort data and we have a dynamics solver)
+      bool has_torques = false;
+      if (!waypoints_[i]->hasEffort() && dynamics_solver)
+      {
+        waypoints_[i]->copyJointGroupPositions(group_->getName(), joint_positions);
+        waypoints_[i]->copyJointGroupVelocities(group_->getName(), joint_velocities);
+        waypoints_[i]->copyJointGroupAccelerations(group_->getName(), joint_accelerations);
```

Comment:
Below there are checks for `has{Velocities,Accelerations}`, what does this do if those are false?

Reply:
Refactored; no longer relevant.